### PR TITLE
Add adios

### PR DESCRIPTION
**Pull request by ax3l**
Create a recipe for ADIOS and its numpy bindings.

https://www.olcf.ornl.gov/center-projects/adios/

Build without MPI support. Only build-able on Unix platforms.
Inspired by the [hdf5-feedstock](https://github.com/conda-forge/hdf5-feedstock) under BSD 3-Clause license :-)

### CODE REVIEW

**Comment by ax3l** [comment]
@pnorbert @jychoi-hpc @isosc I am in the process of trying to add ADIOS1 to conda-forge.
Does anyone of you want to be co-maintainer of the package?

File: recipes/adios/build.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+if [ "$(uname)" == "Darwin" ]; then
+    export CXX="${CXX} -stdlib=libc++"
+    export LDFLAGS="${LDFLAGS} -Wl,-rpath,$PREFIX/lib"
+
+    # remove -lrt
+    sed -i '.bak' 's/ -lrt//g' $SRC_DIR/wrappers/numpy/Makefile
+fi
+
+# Python3 fixes
```
Comment (ax3l) [inline recipes/adios/build.sh]: upstreamed for next release in https://github.com/ornladios/ADIOS/pull/178

File: recipes/adios/build.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+if [ "$(uname)" == "Darwin" ]; then
+    export CXX="${CXX} -stdlib=libc++"
+    export LDFLAGS="${LDFLAGS} -Wl,-rpath,$PREFIX/lib"
+
+    # remove -lrt
```
Comment (ax3l) [inline recipes/adios/build.sh]: upstreamed for next release in https://github.com/ornladios/ADIOS/pull/179

File: recipes/adios/build.sh [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            --with-zlib=$PREFIX \
+            --without-hdf5 \
+            --without-phdf5 \
+            --without-sz \
+            --without-szip \
+            --without-zfp \
+            --without-mpi
+
+# TODO nice to have
+# libtool error:
```
Comment (ax3l) [inline recipes/adios/build.sh]: somehow passes the `FCFLAGS` and all `-L` paths to `libtool`... can't reproduce locally.

File: recipes/adios/build.sh [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+#           --enable-fortran
+# configure error:
```
Comment (ax3l) [inline recipes/adios/build.sh]: note: `hdf5.h` is found but compile test in configure fails.
Can be enabled later on, only interesting for `bp2h5` / `h5tobp` tools.

**Comment by ax3l** [comment]
Ok, the second package might need to be added after the first in order to find the c library :)

File: recipes/adios/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+#   - {{ compiler('fortran') }}
+    - python
+  host:
+    - blosc
+    - bzip2
+    - hdf5 1.10.1
+    - zlib 1.2.11
```
Comment (scopatz) [inline recipes/adios/meta.yaml]: I don't think that these need the hard pinnings if your are using the new compilers.  Maybe @conda-forge/staged-recipes can weigh in.

**Comment by scopatz** [comment]
Thanks for putting this in and congrats on your first recipe to conda-forge!

**Comment by ax3l** [comment]
@scopatz thank you for the review and warm welcome! I removed the pinning of fixed hdf5 & zlib versions and it still works :)

**Comment by scopatz** [comment]
Thanks!  If it fails for some reason, you can add the pins back in later on in the feedstock.

**Pull request merged**